### PR TITLE
Working render layers

### PR DESCRIPTION
**Pull request by maximumpower55**
tldr: the tinted glass part of transoms now render correctly.
Tested on a dedicated server, and everything loaded fine.

### CODE REVIEW

File: src/main/java/io/github/debuggyteam/architecture_extensions/DeferredRegistration.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 public class DeferredRegistration {
 	private static Multimap<Identifier, Entry> deferrals = MultimapBuilder.hashKeys().arrayListValues(2).build();
-	
+
```
Comment (falkreon) [inline src/main/java/io/github/debuggyteam/architecture_extensions/DeferredRegistration.java]: lots of extraneous formatting - intellij is deleting whitespace without telling you again

File: src/main/java/io/github/debuggyteam/architecture_extensions/api/BlockType.java [hunk 2]
```diff
@@ -53,34 +53,36 @@ public enum BlockType {
...
+		return renderLayer;
+	}
+
 	@Override
 	public String toString() {
 		return name().toLowerCase(Locale.ROOT);
 	}
 
 	public TypedGroupedBlock register(BlockGroup group, BlockGroup.GroupedBlock groupedBlock, BlockCreationCallback callback) {
-		return register(group, groupedBlock, callback, ArchitectureExtensions.MOD_CONTAINER.metadata().id()); // If no id is specified, use our own id
```
Comment (falkreon) [inline src/main/java/io/github/debuggyteam/architecture_extensions/api/BlockType.java:74]: I want to retain the modid of the requester for as long as possible for error reporting

File: src/main/java/io/github/debuggyteam/architecture_extensions/resource/DataGeneration.java [hunk 3]
```diff
@@ -25,7 +25,7 @@
 import net.minecraft.util.Identifier;
 
 public final class DataGeneration {
-	static final Set<BlockType.TypedGroupedBlock> BLOCKS = Sets.newHashSet();
+	public static final Set<BlockType.TypedGroupedBlock> BLOCKS = Sets.newHashSet();
```
Comment (falkreon) [inline src/main/java/io/github/debuggyteam/architecture_extensions/resource/DataGeneration.java:28]: Burying the lead here. Is this a thing we want available for API?

File: src/main/java/io/github/debuggyteam/architecture_extensions/util/SafeRenderLayer.java [hunk 4]
```diff
@@ -1,13 +1,33 @@
...
+	CUTOUT(1);
+
+	private final int priority;
+
+	SafeRenderLayer(int priority) {
+		this.priority = priority;
+	}
+
+	@Override
+	public RenderLayer get() {
```
Comment (falkreon) [inline src/main/java/io/github/debuggyteam/architecture_extensions/util/SafeRenderLayer.java:22]: It's shocking that this runs on dedicated server at all. Can we restrict this method to the client environment only?

**Review by falkreon — CHANGES_REQUESTED** [review]
A couple things have my hackles raised about this diff. Let's talk.

File: src/main/java/io/github/debuggyteam/architecture_extensions/util/SafeRenderLayer.java (hunk 4 above)
Comment (maximumpower55) [reply src/main/java/io/github/debuggyteam/architecture_extensions/util/SafeRenderLayer.java:22]: It is, I put a ClientOnly annotation on the interface, that annotation's javadocs specifies that it strips the methods related to the interface

File: src/main/java/io/github/debuggyteam/architecture_extensions/resource/DataGeneration.java (hunk 3 above)
Comment (maximumpower55) [reply src/main/java/io/github/debuggyteam/architecture_extensions/resource/DataGeneration.java:28]: No, but it's needed in the client class

File: src/main/java/io/github/debuggyteam/architecture_extensions/api/BlockType.java (hunk 2 above)
Comment (maximumpower55) [reply src/main/java/io/github/debuggyteam/architecture_extensions/api/BlockType.java:74]: 👍

File: src/main/java/io/github/debuggyteam/architecture_extensions/DeferredRegistration.java (hunk 1 above)
Comment (maximumpower55) [reply src/main/java/io/github/debuggyteam/architecture_extensions/DeferredRegistration.java]: I use vscode not intellij. anyhow I will revert it, I deleted that whitespace by purpose

File: src/main/java/io/github/debuggyteam/architecture_extensions/util/SafeRenderLayer.java (hunk 4 above)
Comment (falkreon) [reply src/main/java/io/github/debuggyteam/architecture_extensions/util/SafeRenderLayer.java:22]: Also thank you for *testing* on a dedicated server. It clearly runs, I just want to stay way on the safe side.

Comment (maximumpower55) [reply src/main/java/io/github/debuggyteam/architecture_extensions/util/SafeRenderLayer.java:22]: I will put a client only annotation on it anyways, so people can clearly see it's for use on the client *only*

**Review by falkreon — APPROVED** [review]
This is a much tighter diff and all my concerns have been addressed. Thanks!

**Pull request merged**